### PR TITLE
Switch CoreHealthMonitor to use MI

### PR DESCRIPTION
**Pull request by dkurepa**
https://dev.azure.com/dnceng/internal/_workitems/edit/6192

### CODE REVIEW

File: src/CoreHealthMonitor/CoreHealthMonitor.cs [hunk 1]
```diff
@@ -172,7 +176,7 @@ await using (FileStream stream = File.OpenRead(file))
 
     private bool IsIgnoredFile(string fileName)
     {
-        string[] patterns = _memoryDumpOptions.CurrentValue.IgnoreDumpPatterns;
+        string[] patterns = _memoryDumpOptions.Value.IgnoreDumpPatterns;
```
Comment (premun) [inline src/CoreHealthMonitor/CoreHealthMonitor.cs:179]: Why the change from options monitor to options?

Comment (dkurepa) [reply src/CoreHealthMonitor/CoreHealthMonitor.cs:179]: We needed OptionMonitors when we were using secrets, because in case of a secret rotation, the new value would get picked up without a deployment. The container url is stable and never changes

Comment (premun) [reply src/CoreHealthMonitor/CoreHealthMonitor.cs:179]: Until it does 😅
I agree that it might not but don't know why we would make it less flexible? Is there some cost to using `OptionsMonitor`?

**Comment by dkurepa** [comment]
after looking into it a bit, it appears this crash dump never actually reported anything in all the years it's been running, Closing this PR to open a new one to remove it

File: src/CoreHealthMonitor/CoreHealthMonitor.cs (hunk 1 above)
Comment (dkurepa) [reply src/CoreHealthMonitor/CoreHealthMonitor.cs:179]: looks like we're not even using that part of the `OptionsMonitor` anywhere. We're basically using it like `IOptions`

Comment (dkurepa) [reply src/CoreHealthMonitor/CoreHealthMonitor.cs:179]: https://alirezafarokhi.medium.com/difference-between-ioptions-ioptionssnapshot-and-ioptionsmonitor-in-asp-netcore-587954bbcea
